### PR TITLE
Upgrade Core to 2.1.3 and Sync to BETA-3.1

### PR DESCRIPTION
**Pull request by cmelchior**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,29 +7,26 @@
...
+* Those were not kept by ProGuard: names of native methods not in the `io.realm.internal` package, names of classes used in method signature (#3596).
+* Missing ProGuard configuration for libraries used by Sync extension (#3596).
+* Error handler was not called when sync session failed (#3597).
+* Permission error when a database file is located at external storage (#3140).
 
 ### Enhancement
 
-* `Realm.compactRealm()` works for encrypted Realms.
+* `Realm.compactRealm()` now works for encrypted Realms.
 *  Added `first(E defaultValue)` and `last(E defaultValue)` methods to `RealmList` and `RealmResult`. These methods will return the provided object instead of throwing an `IndexOutOfBoundsException` if the list is empty.
```
Comment (beeender) [inline CHANGELOG.md]: Extra spaces here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,29 +7,26 @@
...
+* Reduce transformer logger verbosity (#3608).
 *  Added `User.all()` that returns all known Realm Object Server users.
```
Comment (beeender) [inline CHANGELOG.md]: and here.

**Comment by zaki50** [comment]
retest this please

File: dependencies.list [hunk 3]
```diff
@@ -1,3 +1,8 @@
-REALM_SYNC_VERSION=1.0.0-BETA-2.0
-REALM_SYNC_SHA256=c7eb59576b28373283e94dafe42737015657cb0deac435a66c28fc74629ce721
+# Realm Sync Core release used by Realm Java
+# https://github.com/realm/realm-sync/releases
+REALM_SYNC_VERSION=1.0.0-BETA-3.1
+REALM_SYNC_SHA256=43f65bde124589eff9f06ec548816776644fb80498b9aded1fc3b2fd2b9aff5f
+
+# Object Server Release used by Integration tests
+# https://packagecloud.io/realm/realm?filter=debs
 REALM_OBJECT_SERVER_DE_VERSION=1.0.0-BETA-2.1-271
```
Comment (zaki50) [inline dependencies.list]: Why still BETA-2.1?

Comment (zaki50) [inline dependencies.list]: Ah, deb package for BETA-3.1 not yet released..

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,29 +7,26 @@
...
 
-### Enhancements
+### Enhancement
 
+* `Realm.compactRealm()` now works for encrypted Realms.
+* Added `first(E defaultValue)` and `last(E defaultValue)` methods to `RealmList` and `RealmResult`. These methods will return the provided object instead of throwing an `IndexOutOfBoundsException` if the list is empty.
 * Reduce transformer logger verbosity (#3608).
+* Added `User.all()` that returns all known Realm Object Server users.
+
+#Internal
```
Comment (dalinaum) [inline CHANGELOG.md]: `### Internal`? Although some `Internal`s of `CHANGELOG.md` are H1, H2 or H3 randomly, but I think all `Internal`s should be H3 for consistency

**Pull request merged**